transport.go

```diff
@@ -178,10 +192,10 @@ func (t *HTTPTransport) SendEvent(event *Event) {
 		request.Header.Set(headerKey, headerValue)
 	}
 
-	t.wg.Add(1)
+	b := <-t.buffer
```

Comment:
This works as acquiring a lock to access the current `batch`.
A few lines below, `t.buffer <- b` is like releasing the lock.

We must hold the lock during the `select` block below, to guarantee that `b.items` is not closed. Sending on a closed channel panics.

The `select` block "should be quick" (takes a bounded amount of CPU time), because of the `default` case that is executed in case sending on `b.items` would block.